src/api/app/helpers/webui/webui_helper.rb

```diff
@@ -341,5 +341,19 @@ def gravatar_icon(email, size)
   def home_title
     @configuration ? @configuration['title'] : 'Open Build Service'
   end
+
+  def pick_max_problems(failed_checks, build_problems, max_shown)
+    show_checks = [max_shown, failed_checks.length].min
+    show_builds = [max_shown - show_checks, build_problems.length].min
+    # always prefer one build fail
+    if show_builds == 0 && build_problems.present?
+      show_builds += 1
+      show_checks -= 1
+    end
+
+    picked_checks = failed_checks.shift(show_checks)
```

Comment:
`failed_checks` is a [`ActiveRecord::Relation`](https://api.rubyonrails.org/classes/ActiveRecord/Relation.html), it does not know `.shift`.